brave/brave-core
# replace EthProvider/SolProvider P3A metrics, fix provider injection in existing renderers after creating wallet

Resolves https://github.com/brave/brave-browser/issues/30874
Resolves https://github.com/brave/brave-browser/issues/30585
Resolves https://github.com/brave/brave-browser/issues/30873



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Follow test plan for both `Brave.Wallet.EthProvider.4` and `Brave.Wallet.SolProvider.2`. Ensure metric responses are independent of each other.

0) Start with fresh profile. Access brave.com. Ensure metric has a value of 0.
1) Setup the wallet. Reload the brave.com page. Ensure metric has value of 2.
2) Access web3 settings and set the default provider to "Extensions (no fallback)". Reload the page. Ensure the metric has a value of 1.
3) Set the default provider setting to "brave wallet". Reload page. Ensure the metric has a value of 3.
4) Set the default provider setting to "extensions (brave wallet fallback)". Install & setup a third-party extension for the given chain. Reload page. Ensure the metric has a value of 5.
5) Set the default provider setting to "extensions (no fallback)". Reload page. Ensure the metric has a value of 4.
6) Set the default provider setting to "extensions (brave wallet fallback)", reset Brave Wallet. Reload page. Ensure metric has a value of 4.
7) Uninstall the third-party extension. Reload page. Ensure the metric has a value of 0.
